examples/notebooks/models/SPM.ipynb

```diff
@@ -43,33 +43,451 @@
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "negative electrode is of type Uniform1DSubMesh\n",
+      "separator is of type Uniform1DSubMesh\n",
+      "positive electrode is of type Uniform1DSubMesh\n",
+      "negative particle is of type Uniform1DSubMesh\n",
+      "positive particle is of type Uniform1DSubMesh\n",
+      "x_n has 40 mesh points\n",
+      "x_s has 25 mesh points\n",
+      "x_p has 35 mesh points\n",
+      "r_n has 10 mesh points\n",
+      "r_p has 10 mesh points\n"
+     ]
+    }
+   ],
+   "source": [
+    "for k, t in model.default_submesh_types.items():\n",
+    "    print(k,'is of type',t.__name__)\n",
+    "for var, npts in model.default_var_pts.items():\n",
+    "    print(var,'has',npts,'mesh points')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "With these defaults, we can then create our mesh of the given geometry:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mesh = pybamm.Mesh(geometry, model.default_submesh_types, model.default_var_pts)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The next step is to discretise the model equations over this mesh. We do this using the [`pybamm.Discretisation`](https://pybamm.readthedocs.io/en/latest/source/discretisations/discretisation.html) class, which takes both the mesh we have already created, and a dictionary of spatial methods to use for each geometry domain. For the case of the SPM model, we use the following defaults for the spatial discretisation methods:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "macroscale is discretised using FiniteVolume method\n",
+      "negative particle is discretised using FiniteVolume method\n",
+      "positive particle is discretised using FiniteVolume method\n"
+     ]
+    }
+   ],
+   "source": [
+    "for k, method in model.default_spatial_methods.items():\n",
+    "    print(k,'is discretised using',method.__name__,'method')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We then create the `pybamm.Discretisation` object, and use this to discretise the model equations"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "disc = pybamm.Discretisation(mesh, model.default_spatial_methods)\n",
+    "disc.process_model(model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After this stage, all the equations in `model` have been discretised into purely linear algebra expressions that are ready to be evaluated within a time-stepping loop of a given solver. Once again we use the default ODE solver for this model"
```

Comment:
We could show the (simplified) discretised model here?

Reply:
yup, I've put that in now. The `simplify()` didn't work due to dimension problems, but after #253 is resolved I'll put in the simplify.